admin.py

```diff
@@ -216,6 +220,67 @@ def _FetchFromBigstoreREST(self, filename):
     return (result.content, result.status_code)
 
 
+class HistogramsHandler(webapp2.RequestHandler):
+
+  MODEL_CLASS = {
+    'FeatureObserver': models.FeatureObserverHistogram,
+    'MappedCSSProperties': models.CssPropertyHistogram,
+  }
+
+  def _SaveData(self, data, histogram_id):
+    try:
+      model_class = self.MODEL_CLASS[histogram_id]
+    except Exception:
+      logging.error('Invalid Histogram id used: %s' % histogram_id)
+      return
+
+    bucket_id = int(data['bucket_id'])
+    property_name = data['property_name']
+    key_name = '%s_%s' % (bucket_id, property_name)
+
+    # Bucket ID 1 is reserved for number of CSS Pages Visited. So don't add it.
+    if (model_class == models.CssPropertyHistogram and bucket_id == 1):
+      return
+
+    model_class.get_or_insert(key_name,
+      bucket_id=bucket_id,
+      property_name=property_name
+    )
+
+  def get(self):
+    # Attempt to fetch the histograms.xml file from chromium.googlesource.com.
+    result = urlfetch.fetch(HISTOGRAMS_URL)
+
+    if (result.status_code != 200):
+      logging.error('Unable to retrieve chromium histograms.')
+      return
+
+    browsed_histograms = []
+    histograms_content = result.content.decode('base64')
+    dom = xml.dom.minidom.parseString(histograms_content)
+
+    # The histograms.xml file looks like this:
```

Comment:
@ebidel Not sure what you mean. Do you want me to add https://chromium.googlesource.com/chromium/src/+/master/tools/metrics/histograms/histograms.xml?format=TEXT as a comment?

@myakura https://chromium.googlesource.com/chromium/src/+/master/tools/metrics/histograms/histograms.xml?format=TEXT returns the base64 source from what I can see